fetch.bs

```diff
@@ -3342,86 +3345,149 @@ steps:
- "<code>default</code>" or "<code>force-cache</code>", and there is a
- <em>partial</em> <a for=/>response</a> in the HTTP cache for
- <var>httpRequest</var>, modify <var>httpRequest</var>'s
- <a for=request>header list</a> with resume
- <a>headers</a>.
- <!-- XXX xref partial, modify, resume headers -->
+    <ol>
+     <li><p>If <var>storedResponse</var>'s <a for=response>header list</a> contains
+     `<code>ETag</code>`, <a for="header list">append</a> `<code>If-None-Match</code>` with its
+     value to <var>httpRequest</var>'s <a for=request>header list</a>.
+
+     <li><p>If <var>storedResponse</var>'s <a for=response>header list</a> contains
+     `<code>Last-Modified</code>`, <a for="header list">append</a>
+     `<code>If-Modified-Since</code>` with its value to <var>httpRequest</var>'s 
+     <a for=request>header list</a>.
+    </ol>
+
+    <p class=note>See also the 
+    "<a href=https://tools.ietf.org/html/rfc7234#section-4.3.4>Sending a Validation Request</a>"
+    chapter of <cite>HTTP Caching</cite> [[!HTTP-CACHING]].
+
+    <li><p>If the <var>revalidation</var> flag is unset and the <var>partial</var> flag is set, 
+    <a for="header list">append</a> `<code>Range</code>` with an appropriate value to
+    <var>httpRequest</var>'s <a for=request>header list</a>, as per the 
+    "<a href=https://tools.ietf.org/html/rfc7233#section-3.1>Range</a>" chapter of 
+    <cite>HTTP Range Requests</cite> [[!HTTP-RANGE]].
+
+    <li><p>If both the <var>revalidation</var> and <var>partial</var> flags are unset, set
+    <var>response</var> to <var>storedResponse</var>.
+  </ol>
 
  <li>
+  <!-- If response is still null, we require a forward request. -->
   <p>If <var>response</var> is null, run these substeps:
 
   <ol>
    <li><p>If <var>httpRequest</var>'s <a for=request>cache mode</a> is
-   "<code>only-if-cached</code>", then return a
-   <a>network error</a>.
+   "<code>only-if-cached</code>", then return a <a>network error</a>.
 
-   <li><p>Set <var>response</var> to the result of making an
-   <a>HTTP-network fetch</a> using <var>httpRequest</var>
-   with <i>credentials flag</i> if set.
-  </ol>
+   <li><p>Set <var>forwardResponse</var> to the result of making an <a>HTTP-network fetch</a> using
+   <var>httpRequest</var> with <i>credentials flag</i> if set.
 
- <li>
-  <p>If <var>response</var>'s <a for=response>status</a> is <code>304</code> and
-  <var>httpRequest</var>'s <a for=request>cache mode</a> is either
-  "<code>default</code>" or "<code>no-cache</code>", run these substeps:
+   <li><p>If <var>httpRequest</var>'s <var>method</var> is unsafe and <var>forwardResponse</var>'s
+   <a for=response>status</a> begins with <code>2</code> or <code>3</code>, invalidate appropriate
+   stored responses in the HTTP cache, as per the 
+   "<a href=https://tools.ietf.org/html/rfc7234#section-4.4>Invalidation</a>" chapter of 
+   <cite>HTTP Caching</cite> [[!HTTP-CACHING]], and set <var>storedResponse</var> to null.
```

Comment:
I think http://httpwg.org/specs/rfc7231.html#safe.methods would be the most logical. I mostly meant that the term was not referenced. That it's variable is also somewhat problematic, but maybe okay.